src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -558,25 +558,76 @@ def _get_components_data(self, batteries: Set[int]) -> List[InvBatPair]:
 
             inverter_id: int = self._bat_inv_map[battery_id]
 
-            battery_data: Optional[BatteryData] = self._battery_receivers[
-                battery_id
-            ].peek()
-
-            if battery_data is None:
-                _logger.warning("None returned from battery receiver %d.", battery_id)
-                continue
-
-            inverter_data: Optional[InverterData] = self._inverter_receivers[
-                inverter_id
-            ].peek()
-
-            if inverter_data is None:
-                _logger.warning("None returned from inverter receiver %d.", inverter_id)
+            data = self._get_battery_inverter_data(battery_id, inverter_id)
+            if data is None:
+                _logger.warning(
+                    "Skipping battery %d because its message isn't correct.",
+                    battery_id,
+                )
                 continue
 
-            pairs_data.append(InvBatPair(battery_data, inverter_data))
+            pairs_data.append(data)
         return pairs_data
 
+    def _get_battery_inverter_data(
```

Comment:
Maybe it would be nice to have a comment on this method even if it's private to mention that this is dealing mainly with `NaN`s and how they are being handled?

Reply:
Described